zebra/rt_netlink.c

```diff
@@ -1550,7 +1550,7 @@ static ssize_t fill_seg6ipt_encap(char *buffer, size_t buflen,
 	srh->first_segment = segs->num_segs - 1;
 
 	for (i = 0; i < segs->num_segs; i++) {
-		memcpy(&srh->segments[i], &segs->seg[i],
+		memcpy(&srh->segments[segs->num_segs - i - 1], &segs->seg[i],
```

Comment:
This is only 1/2 of the equation, we need to fix the decode side as well.  Add the ipv6 seg6 route from the linux cli and the segments are backwards in comparison.

Reply:
Thanks, fixed,